### PR TITLE
Restore stability for sorting, add tests

### PR DESCRIPTION
**Pull request by ptrfrncsmrph**
Fixes https://github.com/purescript/purescript-arrays/issues/196

### CODE REVIEW

File: src/Data/Array.js [hunk 1]
```diff
@@ -290,7 +290,7 @@ exports.sortByImpl = (function () {
     while (i < mid && j < to) {
       x = xs2[i];
       y = xs2[j];
-      c = fromOrdering(compare(x)(y));
+      c = fromOrdering(compare(x)(y)) || i - j;
```
Comment (kl0tl) [inline src/Data/Array.js:293]: This is equivalent to removing the `else if (c === 0) { /* … */}` later right? I think doing so would be clearer.

**Review by kl0tl — COMMENTED** [review]
Similar changes need to be applied to src/Data/Array/ST.js and test/Test/Data/Array/ST.purs.

Could you also add a comment to sort, sortBy and sortWith in Data.Array and Data.Array.ST to document their stability? I think something along the lines of “Sorting is stable: the order of equal elements is preserved.” should be enough.

**Comment by kl0tl** [comment]
I updated the compiler version specified in the .github/workflows/ci.yml on master, the CI build should pass after a rebase.

File: src/Data/Array.js (hunk 1 above)
Comment (ptrfrncsmrph) [reply src/Data/Array.js]: Simplified here 5524a9fa3d0194523e87e25cf385adbfd3b63a26

**Review by kl0tl — APPROVED** [review]
This looks quite good to me (the build failure is fixed by https://github.com/purescript/purescript-arrays/pull/198 on master), thank you very much!

**Comment by JordanMartinez** [comment]
CI is failing with this error because I forgot to update `group'`'s definition in #164. I'm fixing that mistake in #200:
```
[1/1 UserDefinedWarning] src/Data/Array/NonEmpty.purs:356:1

  356  group' :: forall a. Ord a => NonEmptyArray a -> NonEmptyArray (NonEmptyArray a)
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  
  A custom warning occurred while solving type class constraints:
  
    'group'' is deprecated, use groupAll instead
  
  in value declaration group'

           Src   Lib   All
Warnings   0     0/4   0/4
Errors     1     0     1 
```

**Comment by JordanMartinez** [comment]
CI is now failing with this error. I've submitted purescript/purescript-identity#23 to fix `identity`:
```
* Building project in /home/runner/work/purescript-arrays/purescript-arrays
[1/1 ExtraneousClassMember] bower_components/purescript-identity/src/Data/Identity.purs:90:3

  90    fold1 (Identity x) = x
        ^^^^^^^^^^^^^^^^^^^^^^
  
  fold1 is not a member of type class Data.Semigroup.Foldable.Foldable1
  
  in type class instance
  
    Data.Semigroup.Foldable.Foldable1 Identity

           Src   Lib   All
Warnings   0     0/4   0/4
Errors     0     1     1  
```

**Comment by thomashoneyman** [comment]
@JordanMartinez we also need to remove the fold1 implementation. Sorry for the churn, @ptrfrncsmrph 😅

**Comment by JordanMartinez** [comment]
Finally! Thanks @ptrfrncsmrph!

**Pull request merged**